src/storage/memory_store.ts

```diff
@@ -546,9 +548,16 @@ export class MemoryStore implements IStore {
     if (this.tokens.has(tokenConfig.uid)) {
       throw new Error('Already have this token');
     }
-    this.tokens.set(tokenConfig.uid, tokenConfig);
+
+    const newTokenConfig = { ...tokenConfig }; // Clone the token config to avoid modifying the original object
+    if (!tokens.isHathorToken(newTokenConfig.uid) && !newTokenConfig.version) {
+      // TODO: remove this once we have this configured in the wallet-service. Since any fee token will be in the utxo
+      // all the tokens are deposit
+      newTokenConfig.version = TokenVersion.DEPOSIT;
+    }
+    this.tokens.set(newTokenConfig.uid, newTokenConfig);
```

Comment:
Why do you need this? Can't we have the `version` information in the `tokenConfig` var?